website/source/docs/builders/amazon-ebs-volume.html.md

```diff
@@ -0,0 +1,331 @@
+    templates](/docs/templates/configuration-templates.html) for more info)
+
+-   `instance_type` (string) - The EC2 instance type to use while building the
+    AMI, such as "m1.small".
+
+-   `region` (string) - The name of the region, such as "us-east-1", in which to
+    launch the EC2 instance to create the AMI.
+
+-   `secret_key` (string) - The secret key used to communicate with AWS. [Learn
+    how to set this.](/docs/builders/amazon.html#specifying-amazon-credentials)
+
+-   `source_ami` (string) - The initial AMI used as a base for the newly
+    created machine. `source_ami_filter` may be used instead to populate this
+    automatically.
+
+### Optional:
+
+-   `ebs_volumes` (array of block device mappings) - Add the block
+    device mappings to the AMI. The block device mappings allow for keys:
+
+    -   `device_name` (string) - The device name exposed to the instance (for
+         example, "/dev/sdh" or "xvdh"). Required when specifying `volume_size`.
+    -   `delete_on_termination` (boolean) - Indicates whether the EBS volume is
+        deleted on instance termination
+    -   `encrypted` (boolean) - Indicates whether to encrypt the volume or not
+    -   `iops` (integer) - The number of I/O operations per second (IOPS) that the
+        volume supports. See the documentation on
+        [IOPs](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_EbsBlockDevice.html)
+        for more information
+    -   `no_device` (boolean) - Suppresses the specified device included in the
+        block device mapping of the AMI
+    -   `snapshot_id` (string) - The ID of the snapshot
+    -   `virtual_name` (string) - The virtual device name. See the documentation on
+        [Block Device
+        Mapping](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_BlockDeviceMapping.html)
+        for more information
+    -   `volume_size` (integer) - The size of the volume, in GiB. Required if not
+        specifying a `snapshot_id`
+    -   `volume_type` (string) - The volume type. gp2 for General Purpose (SSD)
+        volumes, io1 for Provisioned IOPS (SSD) volumes, and standard for Magnetic
+        volumes
+    -   `tags` (map) - Tags to apply to the volume. These are retained after the
+        builder completes.
+
+-   `associate_public_ip_address` (boolean) - If using a non-default VPC, public
+    IP addresses are not provided by default. If this is toggled, your new
+    instance will get a Public IP.
+
+-   `availability_zone` (string) - Destination availability zone to launch
+    instance in. Leave this empty to allow Amazon to auto-assign.
+
+-   `ebs_optimized` (boolean) - Mark instance as [EBS
+    Optimized](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSOptimized.html).
+    Default `false`.
+
+-   `enhanced_networking` (boolean) - Enable enhanced
+    networking (SriovNetSupport) on HVM-compatible AMIs. If true, add
+    `ec2:ModifyInstanceAttribute` to your AWS IAM policy.
```

Comment:
You can use enhanced networking on the source instance - although it's uncommon I don't think there is a reason to drop this.